.github/workflows/static_checks_and_tests.yml

```
@@ -0,0 +1,53 @@
+name: static checks and tests
+on:
+  pull_request:
+    branches: [ main ]
+  push:
+    branches: [ main ]
+jobs:
+  shfmt-check:
+    runs-on: ubuntu-latest
+    steps:
+    - name: Checkout repository
+      uses: actions/checkout@v2
```

```suggestion
      uses: actions/checkout@v4
```
update this action to address warning [message](https://github.com/redpanda-data/chaos/actions/runs/10930868018):

> The following actions uses node12 which is deprecated and will be forced to run on node16: actions/checkout@v2. For more info: https://github.blog/changelog/2023-06-13-github-actions-all-actions-will-run-on-node16-instead-of-node12-by-default/